### ISSUE TITLE
0 0 0 1 *  - infinite loop

### ISSUE THREAD

**Issue by evgenru** [InsanusMokrassar/krontab#27]
buildSchedule("0 0 0 1 *").next(DateTime(Date().time))

**Comment by InsanusMokrassar** [comment]
Could you please show an output of `buildSchedule("0 0 0 1 *").next()`?

**Comment by evgenru** [comment]
I can't. infinite loop in next() :)

**Comment by InsanusMokrassar** [comment]
Hello, I have checked with the next code:

```kotlin
suspend fun main() {
    val next = buildSchedule("0 0 0 1 *").next()
    println(next ?.format(DateFormat.DEFAULT_FORMAT))
}
```

And got an output: `Tue, 01 Feb 2022 00:00:00 UTC`

So, looks like there is no any loop here. Could you provide more info (project link or some code sample) to let me help you?

**Comment by evgenru** [comment]
Unfortunately, it has already been removed from the project.
But when I researched, in this place there was dayOfMonth = 2 number:

https://github.com/InsanusMokrassar/krontab/blob/32499d02fbaf9fa27c0c80a7f1bb3110ca570b92/src/commonMain/kotlin/dev/inmo/krontab/internal/NearDateTimeCalculator.kt#L106

02/02/2022...03/02/2022...04/02/2022... and so on in the bexon cycle

**Comment by InsanusMokrassar** [comment]
It looks like correct behaviour. The library will give date/time queue according to the configuration (in your case it is 0 seconds, 0 minutes, 0 hours of each second day in any month). In library there is built-in confersations to the flows and different other tools to work with the result of `next` method. What is your case?

**Comment by evgenru** [comment]
bexon cycle in method next, without return.  method freezes

**Comment by InsanusMokrassar** [comment]
Hello, could you send the target platform, timezone and environment info?

**Comment by InsanusMokrassar** [comment]
@evgenru hi, do you have telegram? I would like to talk to you there for more information

**Comment by InsanusMokrassar** [comment]
The solution for that bug will be available in `0.7.3` version.

**I'd like to attract your attention to the fact, that days in string format of kron is starting since 0**

P.S. Sorry for the delay in fix

**Comment by DVDAndroid** [comment]
Firstly, thanks

Tried build `0.7.3-branch_0.7.3-build118` and I had to minus 1 both day of month and month number
So now January = 0 and first day of month = 0?
